infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -145,11 +161,23 @@ private void handleLrEntryRequest(@Nonnull RequestMsg request,
 
             LogReplicationSinkManager sinkManager = sessionToSinkManagerMap.get(session);
 
-            // If no sink Manager is found, drop the message and log an error
+            // We create a sinkManager for sessions that are discovered while bootstrapping LR. But as topology changes,
+            // we may discover new sessions. At the same time, its possible that the remote Source cluster finds a new
+            // session before the local cluster and sends a request to the local cluster.
+            // Since the two events are async, we wait to receive a new session in the incoming request.
+            // If the incoming session is not known to sessionManager drop the message (until session is discovered by
+            // local cluster), otherwise create a corresponding sinkManager.
```

Comment:
The case where a session will eventually be created is the one where just the topology has changed, is that right?

Reply:
correct

Reply:
can you add this detail in the comment so that it is easier to recall when we revisit the TODO?

Reply:
I had added it, and also enhanced a bit in my last push.
Could you please review the comment again and let me know if its okay now?